java/spi.java.hints/src/org/netbeans/modules/java/hints/spiimpl/Utilities.java

```diff
@@ -1408,7 +1437,35 @@ protected JCCatch catchClause() {
             }
             return super.catchClause();
         }
+        
+        public com.sun.tools.javac.util.List<JCTree> classOrInterfaceOrRecordBodyDeclaration(com.sun.tools.javac.util.Name className, boolean isInterface, boolean isRecord) {
+
+            if (token.kind == TokenKind.IDENTIFIER) {
+                if (token.name().startsWith(dollar)) {
+                    com.sun.tools.javac.util.Name name = token.name();
+
+                    Token peeked = S.token(1);
+
+                    if (peeked.kind == TokenKind.SEMI) {
+                        nextToken();
+                        nextToken();
+
+                        return com.sun.tools.javac.util.List.<JCTree>of(F.Ident(name));
+                    }
+                }
+            }
 
+            com.sun.tools.javac.util.List<JCTree> result = null;
+            Class[] argsType = {com.sun.tools.javac.util.Name.class, boolean.class, boolean.class};
+            try {
+                result = (com.sun.tools.javac.util.List<JCTree>) MethodHandles.lookup().findSpecial(JavacParser.class, "classOrInterfaceOrRecordBodyDeclaration", MethodType.methodType(com.sun.tools.javac.util.List.class, argsType), JackpotJavacParser.class) // NOI18N
+                        .invoke(this, className, false, false);
+            } catch (Throwable ex) {
+                Logger.getLogger(Utilities.class.getName()).log(Level.SEVERE, null, ex);
```

Comment:
I think it should be fine to just fail here, shouldn't it (the method must exist, as the current method overrides it):
```suggestion
                throw new IllegalStateException(ex);
```